src/store/plugins/confirmations.js

```diff
@@ -0,0 +1,33 @@
+import Vue from 'vue';
+import { i18n } from '../../popup/utils/i18nHelper';
+
+const confirmationsText = {
+  login: i18n.t('pages.confirmations.login'),
+  messageSign: i18n.t('pages.confirmations.message-sign'),
+  transactionSign: i18n.t('pages.confirmations.transaction-sign'),
+};
+
+export default store =>
+  store.registerModule('confirmations', {
+    namespaced: true,
+    state: {
+      undeniable: {
+        login: false,
+        messageSign: false,
+        transactionSign: 0,
+      },
+    },
+    getters: {
+      list: ({ undeniable }) =>
+        Object.entries(confirmationsText).map(([name, text]) => ({
+          name,
+          text,
+          value: undeniable[name],
+        })),
+    },
+    mutations: {
+      setConfirmationValue(state, { name, value }) {
+        Vue.set(state.undeniable, name, value);
```

Comment:
why `Vue.set` is necessary? I think it doesn't because all properties of `undeniable` are predefined.